src/test/java/uk/gov/ons/ctp/response/casesvc/service/impl/CaseCreationServiceTest.java

```diff
@@ -95,4 +96,76 @@ public void testCreateCaseAndCaseGroupWithoutChildFromMessage() {
     assertEquals(
         UUID.fromString(sampleUnitParent.getActionPlanId()), capturedCase.get(0).getActionPlanId());
   }
+
+  /**
+   * Create a Case and a Casegroup from the message that would be on the Case Delivery Queue. Child
+   * party present.
+   */
+  @Test
+  public void testCreateCaseAndCaseGroupWithChildFromMessage() {
+
+    SampleUnit sampleUnitChild = new SampleUnit();
+    // Base sample unit data for child
+    sampleUnitChild.setActionPlanId("7bc5d41b-0549-40b3-ba76-42f6d4cf3991");
+    sampleUnitChild.setCollectionInstrumentId("ed0015f0-2e7f-4cf3-ba6f-a752aebaf8a7");
+    sampleUnitChild.setPartyId("73528fd7-ef04-4697-a94c-54edf3e73282");
+    sampleUnitChild.setSampleUnitRef("str1235");
+    sampleUnitChild.setSampleUnitType("BI");
+    sampleUnitChild.setId(UUID.randomUUID().toString());
+    SampleUnitChildren sampleUnitChildren =
+        new SampleUnitChildren(new ArrayList<>(Collections.singletonList(sampleUnitChild)));
+
+    SampleUnitParent sampleUnitParent = new SampleUnitParent();
+    // Parent Only field
+    sampleUnitParent.setSampleUnitChildren(sampleUnitChildren);
+    sampleUnitParent.setCollectionExerciseId("14fb3e68-4dca-46db-bf49-04b84e07e77c");
+    // Base sample unit data for parent
+    sampleUnitParent.setActionPlanId("7bc5d41b-0549-40b3-ba76-42f6d4cf3992");
+    sampleUnitParent.setCollectionInstrumentId("8bae64c5-a282-4e87-ae5d-cd4181ba6c73");
+    sampleUnitParent.setPartyId("7bc5d41b-0549-40b3-ba76-42f6d4cf3992");
+    sampleUnitParent.setSampleUnitRef("str1234");
+    sampleUnitParent.setSampleUnitType("B");
+    sampleUnitParent.setId(UUID.randomUUID().toString());
+
+    caseService.createInitialCase(sampleUnitParent);
+
+    // CaseGroup attributes from parent data, case from child sample unit
+    ArgumentCaptor<CaseGroup> caseGroup = ArgumentCaptor.forClass(CaseGroup.class);
+    verify(caseGroupRepo, times(1)).saveAndFlush(caseGroup.capture());
+
+    List<CaseGroup> capturedCaseGroup = caseGroup.getAllValues();
```

Comment:
you could use getValue() and then you wouldn't need to have all the .get(0). bits lower down

Reply:
Cheers that's much better